src/main/java/org/sunbird/org/service/ExtendedOrgServiceImpl.java

```diff
@@ -58,33 +58,55 @@ public SBApiResponse createOrg(Map<String, Object> request, String userToken) {
-				updateRequest.put(Constants.SB_ORG_ID.toLowerCase(), orgId);
+				String orgName = (String) requestData.get(Constants.ORG_NAME);
 				String orgType = (String) requestData.get(Constants.ORGANIZATION_TYPE);
-				if (requestData.containsKey(Constants.SB_ROOT_ORG_ID)
-						&& StringUtils.isNotEmpty((String) requestData.get(Constants.SB_ROOT_ORG_ID))) {
-					updateRequest.put(Constants.SB_ROOT_ORG_ID, (String) requestData.get(Constants.SB_ROOT_ORG_ID));
-				} else if (!(Constants.STATE.equalsIgnoreCase(orgType)
-						|| Constants.MINISTRY.equalsIgnoreCase(orgType))) {
-					// OrgType is not state / ministry and sbRootOrgId value given as Null... let's
-					// find the rootOrgId
-					String sbRootOrgId = findRootOrgId((String) requestData.get(Constants.ORG_NAME),
-							(String) requestData.get(Constants.MAP_ID));
-					if (StringUtils.isNotEmpty(sbRootOrgId)) {
-						updateRequest.put(Constants.SB_ROOT_ORG_ID, sbRootOrgId);
-					}
+				Map<String, Object> updateRequest = new HashMap<String, Object>();
+				updateRequest.put(Constants.SB_ORG_ID, orgId);
+				updateRequest.put(Constants.ORG_NAME, orgName);
+				updateRequest.put(Constants.SB_ORG_TYPE, orgType);
+				String mapId = (String) requestData.get(Constants.MAP_ID);
+				String orgCode = (String) requestData.get(Constants.ORG_CODE);
+				String sbRootOrgid = (String) requestData.get(Constants.SB_ROOT_ORG_ID);
+				updateRequest.put(Constants.SB_SUB_ORG_TYPE, requestData.get(Constants.ORGANIZATION_SUB_TYPE));
+				if(!StringUtils.isEmpty(mapId)){
+					updateRequest.put(Constants.MAP_ID, mapId);
+				} else {
+					mapId = createMapId(requestData);
+					updateRequest.put(Constants.MAP_ID, mapId);
+					updateRequest.put(Constants.ORG_CODE, mapId);
 				}
-				Map<String, Object> compositeKey = new HashMap<String, Object>() {
-					private static final long serialVersionUID = 1L;
-					{
-						put(Constants.ORG_NAME, requestData.get(Constants.ORG_NAME));
-						put(Constants.MAP_ID, requestData.get(Constants.MAP_ID));
+				if(!StringUtils.isEmpty(orgCode)){
+					updateRequest.put(Constants.ORG_CODE, orgCode);
+				}
+				if (!Constants.STATE.equalsIgnoreCase(orgType) && !Constants.MINISTRY.equalsIgnoreCase(orgType)) {
+					updateRequest.put(Constants.PARENT_MAP_ID, requestData.get(Constants.PARENT_MAP_ID));
+					if(!StringUtils.isEmpty(sbRootOrgid)){
+						updateRequest.put(Constants.SB_ROOT_ORG_ID, sbRootOrgid);
+					} else {
+						updateRequest.put(Constants.SB_ROOT_ORG_ID, fetchRootOrgId(requestData));
 					}
-				};
-				cassandraOperation.updateRecord(Constants.SUNBIRD_KEY_SPACE_NAME, Constants.TABLE_ORG_HIERARCHY,
-						updateRequest, compositeKey);
-				response.getResult().put(Constants.ORGANIZATION_ID, orgId);
-				response.getResult().put(Constants.RESPONSE, Constants.SUCCESS);
+				} else {
+					updateRequest.put(Constants.PARENT_MAP_ID, Constants.SPV);
+				}
+				if (!StringUtils.isEmpty(mapId)){
```

Comment:
This condition will be always true -- since mapId value assigned from createMapId method.... 
This suppose to check for null against requestData.get(Constants.MAP_ID))